addons/delivery/tests/test_packing_delivery.py

```diff
@@ -104,3 +105,56 @@ def test_send_to_shipper_without_sale_order(self):
         picking_ship.action_confirm()
         picking_ship.button_validate()
         self.assertEqual(picking_ship.state, 'done')
+
+    def test_batch_picking_delivery(self):
+        """
+            Check that when Put in Pack is called for batch pickings (i.e. faked with multi-record action
+            calling to avoid extra batch+delivery module for just a test) then:
+             - Same delivery carrier = works
+             - Different delivery carriers = UserError
+        """
+        self.env['stock.quant']._update_available_quantity(self.productA, self.stock_location, 4.0)
+        another_test_carrier_product = self.env['product.product'].create({
+            'name': 'Another test carrier product',
+            'type': 'service',
+        })
+        test_carrier = self.env['delivery.carrier'].create({
+            'name': 'Another Test carrier',
+            'delivery_type': 'fixed',
+            'product_id': another_test_carrier_product.id,
+        })
+
+        delivery_1 = self.env['stock.picking'].create({
+            'partner_id': self.env['res.partner'].create({'name': 'A partner'}).id,
+            'picking_type_id': self.warehouse.out_type_id.id,
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'carrier_id': self.test_carrier.id
+        })
+        move_1 = self.env['stock.move.line'].create({
+            'product_id': self.productA.id,
+            'product_uom_id': self.productA.uom_id.id,
+            'picking_id': delivery_1.id,
+            'qty_done': 1,
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id
+        })
+
+        # Test that differing carrier + put in pack = UserError
+        delivery_2 = delivery_1.copy({'carrier_id': test_carrier.id})
+        move_1.copy({'picking_id': delivery_2.id, 'qty_done': 1})
+        # recreate the `action_put_in_pack`` steps so we don't have to add test to new module for batch pickings
+        # to use batch version of method (which bypass the ensure_one() check in the stock_picking action)
+        move_lines_to_pack = (delivery_1 | delivery_2)._package_move_lines()
+        self.assertEqual(len(move_lines_to_pack), 2, 'There should exist moves to be able to put in pack')
```

Comment:
I think my brain was melting when I wrote these tests. I reworded the sentence because it was grammatically a mess and made sure to make it "move lines" :wink: